spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Module.scala

```diff
@@ -38,6 +39,11 @@ object Module {
     CaffeLoader.load[T](model, defPath, modelPath, matchAll)
   }
 
+  def loadCaffeDynamic[T: ClassTag](defPath: String, modelPath: String, matchAll: Boolean = true)(
```

Comment:
@yiheng  seems  overloaded methods won't work if they have defined default argument even they have different para types , that's why I named another one.